presto-main/src/main/java/com/facebook/presto/operator/DeleteOperator.java

```diff
@@ -142,8 +149,11 @@ public Page getOutput()
         }
         state = State.FINISHED;
 
-        Collection<Slice> fragments = pageSource().commit();
+        UpdatablePageSource updatablePageSource = pageSource();
+        Collection<Slice> fragments = updatablePageSource.commit();
         committed = true;
+        systemMemoryUsageListener.updateSystemMemoryUsage(updatablePageSource.getDeltaMemory() - bufferBytes);
```

Comment:
Call `.getDeltaMemory()` once and save the result. In theory, it could change between calls